src/prof_sys.c

```diff
@@ -475,8 +480,73 @@ prof_dump_read_maps_cb(void *read_cbopaque, void *buf, size_t limit) {
 	return malloc_read_fd(mfd, buf, limit);
 }
 
+#else // __APPLE__
+#ifdef __LP64__
+typedef struct mach_header_64 je_dyld_header;
+typedef struct segment_command_64 je_dyld_segment_command;
+
+#define JE_DYLD_IMAGE_MAGIC     MH_MAGIC_64
+#define JE_DYLD_IMAGE_MAGIC_INV MH_CIGAM_64
```

Comment:
Maybe `_REVERSE` or stick to `_CIGAM`? The `_INV` feels a bit close to `invalid`.